packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/advanced_marker.dart

```diff
@@ -0,0 +1,171 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show Offset;
+import 'package:flutter/foundation.dart';
+
+import '../../google_maps_flutter_platform_interface.dart';
+
+/// Marks a geographical location on the map.
+///
+/// Extends [Marker] and provides additional features.
+@immutable
+class AdvancedMarker extends Marker {
+  /// Creates a set of marker configuration options.
+  ///
+  /// Specifies a marker with a given [collisionBehavior]. Default is
+  /// [MarkerCollisionBehavior.required].
+  const AdvancedMarker({
+    required super.markerId,
+    super.alpha,
+    super.anchor,
+    super.consumeTapEvents,
+    super.draggable,
+    super.flat,
+    super.icon,
+    super.infoWindow,
+    super.position,
+    super.rotation,
+    super.visible,
+    super.zIndex,
```

Comment:
I can't remember if this came up earlier in the review: Can we make this `int zIndex` and have it set the super `zIndex` to the int value in the constructor, so that we aren't replicating the API mistake (https://github.com/flutter/flutter/issues/155897) to this new class?

Reply:
It's possible to change this to `int zIndex` but there's a problem with changing double to int in `copyWith`. Updated `copyWith` is not a valid override of `Marker`'s `copyWith` where `zIndex` has to be double. 

One option is to keep `double? zIndex` in `copyWith` but it won't be consistent with `AdvancedMarker`'s constructor. 

Adding a new method like `copyObjectWith` doesn't sound right. Original `copyWith` will still be available and will return an instance of `Marker` when called on `AdvancedMarker`

Reply:
Hm. I think the best option is probably to make it `int zIndex` in the constructor, and leave `copyWith` wrong, and just truncate in there. `copyWith` will be semi-fixed when we add the new int variant of zIndex to the base class (and thus to `copyWith`). Not ideal, but since we're inheriting it's probably the best option we have without a breaking change.

Reply:
I've made this change. There's a `.toInt()` call inside `copyWith` to convert from Marker's double to AdvancedMarker's int